##### In this notebook, SQL is used to analyze the final analytical dataset and answer key business questions related to pizza sales performance. The objective is to uncover meaningful insights into customer purchasing behavior, product performance, and revenue trends to support data-driven decision-making.


###  Importing Libraries

In [74]:
import pandas as pd
from sqlalchemy import create_engine
import warnings
warnings.filterwarnings('ignore')

### Creating Database Connection

In [75]:
engine = create_engine('sqlite:///pizza_sales_analysis.db')
df = pd.read_sql_query('select * from pizza',engine)

### Business Questions

#### Overall Performance
##### What is the total number of orders, pizzas sold, and total revenue?

In [76]:
pd.read_sql_query('select count(distinct order_id) as orders ,sum(quantity) as quantity,sum(revenue) as revenue from pizza',engine)

,orders,quantity,revenue
0,21350,49574,817860.05


##### The business recorded 21,350 total orders, resulting in the sale of 49,574 pizzas and generating a total revenue of $817,860.05. This indicates strong overall sales performance and provides a solid foundation for deeper analysis of product demand, customer behavior, and revenue trends.

#### Product Performance

##### Which pizza sells the most by quantity?

In [77]:
pd.read_sql_query('select name,sum(quantity) as quantity from pizza group by name order by quantity desc  limit 10',engine)

,name,quantity
0,The Classic Deluxe Pizza,2453
1,The Barbecue Chicken Pizza,2432
2,The Hawaiian Pizza,2422
3,The Pepperoni Pizza,2418
4,The Thai Chicken Pizza,2371
5,The California Chicken Pizza,2370
6,The Sicilian Pizza,1938
7,The Spicy Italian Pizza,1924
8,The Southwest Chicken Pizza,1917
9,The Big Meat Pizza,1914


##### Which pizza sells the least?

In [87]:
pd.read_sql_query('select name,sum(quantity) as most_sold  from pizza group by name order by most_sold limit 1 ',engine)


,name,most_sold
0,The Brie Carre Pizza,490


##### Which pizza generates the most revenue?

In [78]:
pd.read_sql_query('select name,sum(revenue) as revenue from pizza group by name order by revenue desc  limit 10',engine)

,name,revenue
0,The Thai Chicken Pizza,43434.25
1,The Barbecue Chicken Pizza,42768.00
2,The California Chicken Pizza,41409.50
3,The Classic Deluxe Pizza,38180.50
4,The Spicy Italian Pizza,34831.25
5,The Southwest Chicken Pizza,34705.75
6,The Italian Supreme Pizza,33476.75
7,The Hawaiian Pizza,32273.25
8,The Four Cheese Pizza,32265.70
9,The Sicilian Pizza,30940.50


##### The Classic Deluxe Pizza is the most sold pizza by quantity.
##### The Thai Chicken Pizza is the highest revenue-generating pizza.
##### The top-selling pizzas by quantity and revenue are not exactly the same, showing the impact of pricing and product mix.
##### Sales are well distributed across multiple pizzas, indicating balanced customer demand rather than reliance on a single product.

#### Size Analysis

##### How does revenue vary by pizza size?

In [97]:
pd.read_sql_query('select size,sum(revenue) as revenue from pizza group by size order by revenue desc ',engine)

,size,revenue
0,L,375318.70
1,M,249382.25
2,S,178076.50
3,XL,14076.00
4,XXL,1006.60


##### What is the average order quantity by pizza size?

In [82]:
pd.read_sql_query('select size,avg(quantity) as avg_quantity from pizza group by size ',engine)


,size,avg_quantity
0,L,1.023211
1,M,1.016250
2,S,1.018816
3,XL,1.014706
4,XXL,1.000000


##### Large (L) pizzas generate the highest revenue ($375,318.70), contributing a significant share of total sales.
##### Medium and Small sizes follow, while XL and XXL contribute very little revenue, indicating low demand for extra-large sizes.
##### The average quantity per order is close to 1 for all sizes, showing customers typically buy one pizza per order item.

#### Category Analysis

##### Which pizza category sells the most by quantity?

In [83]:
pd.read_sql_query('select category,sum(quantity) as salles  from pizza group by category order by salles desc ',engine)


,category,salles
0,Classic,14888
1,Supreme,11987
2,Veggie,11649
3,Chicken,11050


##### Which pizza category generates the highest revenue?

In [98]:
pd.read_sql_query('select category,sum(revenue) as rev  from pizza group by category order by rev desc ',engine)

,category,rev
0,Classic,220053.10
1,Supreme,208197.00
2,Chicken,195919.50
3,Veggie,193690.45


##### Classic category sells the most pizzas (14,888 units) and also generates the highest revenue ($220,053.10).
##### Supreme, Chicken, and Veggie categories perform similarly in revenue, showing balanced demand across categories.

#### Time-Based Analysis

##### How does revenue vary by hour of the day?

In [99]:
pd.read_sql_query('select hour,sum(revenue) as revenue  from pizza group by hour order by revenue desc ',engine)


,hour,revenue
0,12,111877.90
1,13,106065.70
2,18,89296.85
3,17,86237.45
4,19,72628.90
5,16,70055.40
6,14,59201.40
7,20,58215.40
8,15,52992.30
9,11,44935.80


##### Sales peak during lunch (12–1 PM) and evening (5–7 PM) hours.
##### Revenue drops significantly after 10 PM, with minimal late-night sales.
##### This clearly highlights lunch and dinner as the busiest business periods.

##### How does revenue vary by month?

In [100]:
pd.read_sql_query('select month_name,sum(revenue) as revenue  from pizza group by month_name order by revenue desc ',engine)


,month_name,revenue
0,July,72557.90
1,May,71402.75
2,March,70397.10
3,November,70395.35
4,January,69793.30
5,April,68736.80
6,August,68278.25
7,June,68230.20
8,February,65159.60
9,December,64701.15


##### Revenue is fairly consistent across months, showing stable demand throughout the year.
##### July and May are the highest revenue months, while October and September show slightly lower performance.

#### Customer Behavior

##### How do weekend sales compare to weekday sales?

In [101]:
pd.read_sql_query('select weekday, sum(revenue) as revenue  from pizza   group by weekday  order by revenue desc ',engine)


,weekday,revenue
0,Friday,136073.90
1,Thursday,123528.50
2,Saturday,123182.40
3,Wednesday,114408.40
4,Tuesday,114133.80
5,Monday,107329.55
6,Sunday,99203.50


##### Friday generates the highest revenue, followed by Thursday and Saturday.
##### Sunday has the lowest revenue, suggesting weaker demand at the end of the weekend.
##### Overall, sales are stronger toward the end of the workweek.